src/taosdump.c

```diff
@@ -7327,12 +7372,22 @@ static int64_t dumpInAvroDataImpl(
                         break;
                     case TSDB_DATA_TYPE_JSON:
                     case TSDB_DATA_TYPE_NCHAR:
-                        if (field->type != TSDB_DATA_TYPE_NCHAR) {
-                            warnPrint("field[%d] type is not nchar/json!\n", i);
-                            bind->is_null = &is_null;
+                        // RecordSchema bytes only covert to nchar type
+                        if (field->type == TSDB_DATA_TYPE_NCHAR) {
+                            dumpInAvroDataNChar(field, &field_value, bind, &is_null);    
                         } else {
-                            dumpInAvroDataNChar(field, &field_value,
-                                    bind, &is_null);
+                            warnPrint("field[%d] type is not nchar/json! field->type=%d\n", i, field->type);
+                            bind->is_null = &is_null;                            
+                        }
+                        break;
+                    case TSDB_DATA_TYPE_VARBINARY:
+                    case TSDB_DATA_TYPE_GEOMETRY:
+                        // RecordSchema bytes only covert to nchar type
+                        if (field->type == TSDB_DATA_TYPE_NCHAR) {
+                            dumpInAvroDataBytes(field, &field_value, bind, &is_null);
+                        } else {
+                            warnPrint("field[%d] type is not varbinary/geometry! field->type=%d\n", i, field->type);
```

Comment:
The error description here does not match the above conditional judgment

Reply:
the same reason with first problem